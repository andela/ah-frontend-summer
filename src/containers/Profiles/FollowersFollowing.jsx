import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchUserProfile } from '../../store/actions/async/profiles';
import FollowerFollowing from '../../components/Profiles/FollowerFollowing';
import PropTypes from 'prop-types';

export class FollowersFollowing extends Component {
    componentDidMount = () => {
        const { 
            match: { params: { username } } = { params: {} }, 
            fetchUserProfile 
        } = this.props;
        const token = localStorage.getItem("token");
        fetchUserProfile({ username, token });
    };

    render() {
        return <FollowerFollowing {...this.props} />;
    }
}

const mapStateToProps = state => {
    return {
        profile: state.profile.profile
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        fetchUserProfile: payload => dispatch(fetchUserProfile(payload))
    };
};

FollowersFollowing.propTypes = {
    fetchUserProfile: PropTypes.func.isRequired,
    profile: PropTypes.object.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(FollowersFollowing);
