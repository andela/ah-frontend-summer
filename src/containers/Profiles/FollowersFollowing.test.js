import React from 'react';
import { shallow } from 'enzyme';
import { FollowersFollowing, mapDispatchToProps } from './FollowersFollowing';

describe("FollowersFollowing", () => {
    const fetchUserProfileMock = jest.fn();
    const userProfileProps = {
        fetchUserProfile: fetchUserProfileMock,
        match: { params: { username: "" }},
        location: { pathname: "" },
        profile: {},
    };
    const followersFollowing = shallow(<FollowersFollowing {...userProfileProps}/>);
    const followersFollowingInstance = followersFollowing.instance();

    it("renders correctly", () => {
        expect(followersFollowing).toMatchSnapshot();
    });
    
    it("componentDidMount calls dispatches actions to fetch user profile", () => {
        followersFollowingInstance.componentDidMount();
        expect(fetchUserProfileMock).toHaveBeenCalled();
    });

    it("mapDispatchToProps", () => {
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).fetchUserProfile();
        expect(dispatch.mock.calls.length).toBe(1);
    });
});
