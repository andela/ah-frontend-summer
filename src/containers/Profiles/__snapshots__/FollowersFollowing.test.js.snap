// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowersFollowing renders correctly 1`] = `
<FollowerFollowing
  fetchUserProfile={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "token": null,
            "username": "",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  location={
    Object {
      "pathname": "",
    }
  }
  match={
    Object {
      "params": Object {
        "username": "",
      },
    }
  }
  profile={Object {}}
/>
`;
